paddle/fluid/framework/data_set.h

```diff
@@ -184,13 +194,16 @@ class DatasetImpl : public Dataset {
   bool keep_unmerged_ins_;
   int min_merge_size_;
   std::vector<std::string> merge_slots_list_;
+  bool fea_eval_ = false;
 };
 
 // use std::vector<MultiSlotType> or Record as data type
 class MultiSlotDataset : public DatasetImpl<Record> {
  public:
   MultiSlotDataset() {}
   virtual void MergeByInsId();
+  virtual void SlotsShuffle(const std::set<uint16_t>& slots_to_replace);
```

Comment:
you should consider user pass a string name of slot。 
it's convenient for user to use "var.name", not a int

Reply:
to do

Reply:
changed to std::string